content/en/blog/2022/how-to-get-started-with-opentelemetry-go.md

```diff
@@ -0,0 +1,385 @@
+
+There are two ways to instrument our app – manually and automatically:
+
+- **Auto instrumentation:** Automatically create spans from the application libraries we use with ready-to-use OpenTelemetry libraries.
+- **Manual instrumentation:** Manually add code to your application to define the beginning and end of each span and the payload.
+  To understand more of the OpenTelemetry jargon, [visit the official documentation](https://opentelemetry.io/docs/concepts/data-sources/).
+
+## Hello world: OpenTelemetry Go example
+
+We will start by creating our to-do service and installing two libraries (Gin and Mongo) to understand how instrumentations work.
+
+### Step 1: Create main.go file for our to-do app
+
+1. Install Gin and Mongo-driver
+
+```go
+go get -u github.com/gin-gonic/gin
+go get go.mongodb.org/mongo-driver/mongo
+```
+
+2. Set up gin and mongo to listen on “/todo”
+
+3. Create some to-do’s to seed Mongo
+
+```go
+package main
+import (
+"context"
+"net/http"
+"github.com/gin-gonic/gin"
+"go.mongodb.org/mongo-driver/bson"
+"go.mongodb.org/mongo-driver/mongo"
+"go.mongodb.org/mongo-driver/mongo/options"
+)
+var client *mongo.Client
+func main() {
+connectMongo()
+setupWebServer()
+}
+func connectMongo() {
+opts := options.Client()
+opts.ApplyURI("mongodb://localhost:27017")
+client, \_ = mongo.Connect(context.Background(), opts)
+//Seed the database with todo's
+docs := []interface{}{
+bson.D{{"id", "1"}, {"title", "Buy groceries"}},
+bson.D{{"id", "2"}, {"title", "install Aspecto.io"}},
+bson.D{{"id", "3"}, {"title", "Buy dogz.io domain"}},
+}
+client.Database("todo").Collection("todos").InsertMany(context.Background(), docs)
+}
+func setupWebServer() {
+r := gin.Default()
+r.GET("/todo", func(c *gin.Context) {
+collection := client.Database("todo").Collection("todos")
+//Important: Make sure to pass c.Request.Context() as the context and not c itself - TBD
+cur, findErr := collection.Find(c.Request.Context(), bson.D{})
+if findErr != nil {
+c.AbortWithError(500, findErr)
+return
+}
+results := make([]interface{}, 0)
+curErr := cur.All(c, &results)
+if curErr != nil {
+c.AbortWithError(500, curErr)
+return
+}
+c.JSON(http.StatusOK, results)
+})
+\_ = r.Run(":8080")
+}
+```
+
+Now that our small todo app is ready, let’s introduce OpenTelemetry.
+
+### Step 2: Install OpenTelemetry Go
+
+We will be configuring OpenTelemetry to instrument our Go app.
+
+1. To install the OTel SDK, run:
+
+```go
+go get go.opentelemetry.io/otel /
+go.opentelemetry.io/otel/sdk /
+```
+
+2. Instrument our Gin and Mongo libraries to generate traces.
+
+3. Gin instrumentation: Install otelgin
+
+```go
+go get go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin
+```
+
+4. Mongo instrumentation: Install otelmongo
+
+```go
+go get go.opentelemetry.io/contrib/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo
+```
```

Comment:
I'd recommend collapsing this into a single call to `go get` than 4 separate steps for people to write.

Also the language chosen for the code blocks should be `console` and not `go`.